drivers/scheduler/k8s/k8s.go

```diff
@@ -361,9 +364,16 @@ func (k *k8s) WaitForRunning(ctx *scheduler.Context) error {
 	return nil
 }
 
-func (k *k8s) Destroy(ctx *scheduler.Context) error {
+func (k *k8s) Destroy(ctx *scheduler.Context, opts map[string]bool) error {
+	var podList []v1.Pod
 	for _, spec := range ctx.App.SpecList {
 		if obj, ok := spec.(*apps_api.Deployment); ok {
+			if pods, err := k8sutils.GetDeploymentPods(obj); err != nil {
+				logrus.Warnf("[%v] Error getting deployment pods. Err: %v", ctx.App.Key, err)
+			} else {
+				podList = append(podList, pods...)
```

Comment:
Looks like we need the `podList` only if `opts[scheduler.OptionsWaitForResourceLeakCleanup]` is set. So we avoid the fetching of pods for other use cases.